hypershift-operator/controllers/nodepool/nodepool_controller.go

```diff
@@ -1429,6 +1576,14 @@ func (r *NodePoolReconciler) enqueueNodePoolsForConfig(obj client.Object) []reco
 				break
 			}
 		}
+		for _, v := range nodePoolList.Items[key].Spec.TunedConfig {
```

Comment:
having this here will needlessly run the loop even when the result was found in the loop above via .Spec.Config so we'll want to return earlier.

Reply:
Good catch, I rearranged this part of the function to avoid unnecessary looping when we already know a NodePool needs to be reconciled.